runtime/complex.go

```diff
@@ -28,11 +30,12 @@ var ComplexType = newBasisType("complex", reflect.TypeOf(Complex{}), toComplexUn
 type Complex struct {
 	Object
 	value complex128
+	hash  int
```

Comment:
I question whether caching the hash value has much value. Note that CPython does not cache it for complex or float. Perhaps we should remove the caching in float until there's a good reason to add it.

Note that strings are good candidates for caching since a) the length of the string can be very long (amortizes the runtime of hash(s)) and b) many commonly used strings (e.g. attribute names) are interned meaning that the cache is highly effective.

Neither of these factors apply to float or complex.